### PR TITLE
Interop gen additional literal types

### PR DESCRIPTION
**Pull request by sagar-h007**
This PR improves interop_gen support for TypeScript literal types that aren’t represented as simple literals in the AST (for example, prefix unary expressions like -1).

When a literal can’t be handled directly, the generator now falls back to the TypeScript TypeChecker to resolve the type. If the resolved type is a literal, its value is used; otherwise, the code safely falls back to the underlying Dart type, preserving existing behavior.

The change also adds the minimal TypeScript bindings needed for this resolution and includes an integration test to cover the new case and prevent regressions.

Fixes:#465

---

- I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

	•	This is a small, targeted change based on an existing issue.
	•	The code follows the Dart style guide and is formatted with dart format.
	•	New integration tests were added, and all existing tests pass locally.
	•	No changelog or pubspec update is needed for this internal tooling change.

</details>

### CODE REVIEW

File: web_generator/lib/src/interop_gen/transform/transformer.dart [hunk 1]
```diff
@@ -1742,34 +1742,96 @@ class Transformer {
...
+                    kind: LiteralKind.$true,
+                    value: true,
+                  );
+                } else if (typeStr == 'false') {
+                  return LiteralType(
+                    isNullable: isNullable ?? false,
+                    kind: LiteralKind.$false,
+                    value: false,
+                  );
+                }
```
[the remark this answers is not in the document]
Comment (sagar-h007) [reply web_generator/lib/src/interop_gen/transform/transformer.dart]: I’ll simplify this block to reduce duplication.

File: web_generator/test/integration/interop_gen/literal_test_input.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export declare const negativeOne: -1;
+// Binary expressions in type position are not valid TS in d.ts files usually.
```
Comment (srujzs) [inline web_generator/test/integration/interop_gen/literal_test_input.d.ts]: Do we have an example we can test where it *is* valid?

File: web_generator/lib/src/js/typescript.types.dart [hunk 3]
```diff
@@ -15,6 +15,69 @@ import 'annotations.dart';
 import 'helpers.dart';
 import 'typescript.dart';
 
+extension type const TSTypeFlags._(int _) implements int {
```
Comment (srujzs) [inline web_generator/lib/src/js/typescript.types.dart]: nit: I think I prefer omitting unused constants with maybe a TODO to add more.

File: web_generator/test/integration/interop_gen/literal_test_input.d.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+// export declare const binaryAdd: 1 + 2;
+// export declare const shiftLeft: 1 << 16;
+// export declare const complex: 1 + 2 * 3;
+export declare const existingLiteral: 1;
+export declare const existingString: "hello";
+export declare const existingBoolean: true;
```
Comment (srujzs) [inline web_generator/test/integration/interop_gen/literal_test_input.d.ts]: Let's add a test for `false` and `null` as well.

File: web_generator/lib/src/interop_gen/transform/transformer.dart [hunk 5]
```diff
@@ -1742,34 +1742,90 @@ class Transformer {
...
+                  return LiteralType(
+                    isNullable: isNullable ?? false,
+                    kind: value ? LiteralKind.$true : LiteralKind.$false,
+                    value: value,
+                  );
+                }
+              }
+
+              // Fallback to underlying type if not a literal
+              final underlyingTypeNode = typeChecker.typeToTypeNode(
```
Comment (srujzs) [inline web_generator/lib/src/interop_gen/transform/transformer.dart]: Is there a test exercising this case?

Comment (sagar-h007) [reply web_generator/lib/src/interop_gen/transform/transformer.dart]: Not yet ,. I’ll add a test to explicitly exercise this fallback path.

File: web_generator/lib/src/js/typescript.types.dart (hunk 3 above)
Comment (sagar-h007) [reply web_generator/lib/src/js/typescript.types.dart]: I’ll drop the unused constants for now and leave a TODO for adding more when we actually need them. That should keep things simpler and safer.

**Review by srujzs — COMMENTED** [review]
Mostly LGTM % my comments.

File: web_generator/lib/src/interop_gen/transform/transformer.dart (hunk 5 above)
Comment (srujzs) [reply web_generator/lib/src/interop_gen/transform/transformer.dart]: I might have missed it, but which test case covers this?

**Comment by srujzs** [comment]
Ack, I think the only thing left is the uncomment (and generate the expectation file) and we can go ahead and land this.

**Comment by sagar-h007** [comment]
@srujzs Done — re-enabled the test using a 1n BigInt to hit the fallback path and updated the expected output. All tests are passing.
